### ISSUE TITLE
Mono mkbundle output executable throws: “Error mapping file: No error”

### ISSUE THREAD

**Issue by dns** [mono/mono#8225]
I wrote simple "hello world" console program in C#, and build it using VS2013 (I also try to build it using mono "csc" but the error is same).

```
> mkbundle --sdk "C:\Program Files\Mono" --deps -o test.exe ConsoleApplication2.exe
Using runtime: C:\Program Files\Mono\bin\mono
     Assembly: D:\SANDBOX\Silverlight\ConsoleApplication2\ConsoleApplication2\bin\Release\ConsoleApplication2.exe
       Config: D:\SANDBOX\Silverlight\ConsoleApplication2\ConsoleApplication2\bin\Release\ConsoleApplication2.exe.config
     Assembly: C:\Program Files\Mono\lib\mono\4.5\mscorlib.dll
     Assembly: C:\Program Files\Mono\lib\mono\4.5\I18N.West.dll
     Assembly: C:\Program Files\Mono\lib\mono\4.5\I18N.dll
Generated test.exe
```

The new "test.exe" need mono-2.0-sgen.dll, so I copy it to "C:\Program Files\Mono\bin" and then run test.exe, but the new executable throw a error message to stdout:

```
> test.exe
Error mapping file: No error
```

**Version Used**:
- Mono 5.12.0.226.x64
- Win10 (64-bit).

**Comment by marek-safar** [comment]
@lateralusX does this look familiar?

**Comment by lateralusX** [comment]
Nothing that I have seen before. This mkbundle execution use the --sdk feature that packages bundles differently into the final binary not using the installed C toolchain. Instead it appends bundles directly to a copy of the mono binary (that in this case is not a static linked version, but depending on mono-2.0-sgen.dll). The error seems to occurs when the runtime tries to load (memory mapping the bundles) the bundles. If this is still an issue I could add it to my list and see if I could repro the problem.

**Comment by jaykrell** [comment]
Get us the .exe to debug?

**Comment by jaykrell** [comment]
Or run it along with sysinternals sysmon or such, watch the file opens for errors, near the end. Is something like what I'd do.

**Comment by dns** [comment]
@jaykrell, Here [out.zip](https://github.com/mono/mono/files/2039956/out.zip)

**Comment by jaykrell** [comment]
It is like I thought from code inspection.
mkbundle aligns things to 4k but 64k alignment is needed.

KERNEL32!MapViewOfFileStub:
00007ffc`b3d1a930 48ff2501d80500  jmp     qword ptr [KERNEL32!_imp_MapViewOfFile (00007ffc`b3d78138)] ds:00007ffc`b3d78138={KERNELBASE!MapViewOfFile (00007ffc`b25c4a40)}
0:000> r r9
r9=0000000000027000
0:000> gu
Breakpoint 12 hit
KERNELBASE!MapViewOfFile:
00007ffc`b25c4a40 4c8bdc          mov     r11,rsp
0:000> gu
mono_2_0_sgen!mono_counters_sample+0x248c24:
00007ffc`863df0c0 488bc8          mov     rcx,rax
0:000> r rax
rax=0000000000000000
0:000> !gle
LastErrorValue: (Win32) 0x46c (1132) - The base address or the file offset specified does not have the proper alignment.
LastStatusValue: (NTSTATUS) 0xc0000220 - {Mapped View Alignment Incorrect}  An attempt was made to map a view of a file, but either the specified base address or the offset into the file were not aligned on the proper allocation granularity.

**Comment by jaykrell** [comment]
Here is how you debug this:
xcopy debuggers to bin (or elsewhere)
\bin\amd64\cdb foo.exe
!dlls
see .exe entry point
bp on that
g
bm kern*!*mapviewoffile*
g
look at parameters: first four are rcx rdx r8 r9 and r9 is what you  want
gu 
r rax to see return value
!gle for getlasterror

**Comment by elitest** [comment]
@jaykrell Is this fix available in the [nightlies](https://download.mono-project.com/archive/nightly/windows-installer/)?  Or is the only way to get this is to build from master.  Seems still present in nightlies, but those might be built from the release branch.  Is it possible to download an artifact from a recent jenkins build?

**Comment by jaykrell** [comment]
The fix was never approved.

**Comment by jaykrell** [comment]
https://github.com/mono/mono/pull/8884

**Comment by elitest** [comment]
Thanks.  That is unfortunate.  I will keep an eye on it, it the mean time I'll try my hand at compiling your branch.  I haven't had luck with compiling so far.

**Comment by jaykrell** [comment]
Can you try nightlies though, and check for a better error message?

**Comment by elitest** [comment]
I'm AFK for a week.  Next time I have a windows computer accessible I'll give it a shot.

**Comment by T0bi-Ethirbirge** [comment]
Error mapping file: mono_file_map_error failed file:C:\Users\tobi\Desktop\Test\Test\bin\Debug\New.exe length:0x46D000 offset:0x27000 function:MapViewOfFile error:The base address or the file offset specified does not have the proper alignment.(0x46C).
Now how could we fix this.

**Comment by lateralusX** [comment]
Looks like the PR fixing this issue was merged in September #8884, closing this issue.
